src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractInternalsVisibleToCompletionProvider : CommonCompletionProvider
+    {
+        private const string ProjectGuidKey = nameof(ProjectGuidKey);
+
+        protected abstract IImmutableList<SyntaxNode> GetAssemblyScopedAttributeSyntaxNodesOfDocument(SyntaxNode documentRoot);
+        protected abstract SyntaxNode GetConstructorArgumentOfInternalsVisibleToAttribute(SyntaxNode internalsVisibleToAttribute);
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '\"';
```

Comment:
It seems to me that `CommonCompletionUtilities.IsStartingNewWord` does not help in this situation. It explicitly says [here](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/Completion/CommonCompletionUtilities.cs#L58):
```
// Only want to trigger if we're the first character in an identifier.  If there's a
// character before or after us, then we don't want to trigger.
```
So this would trigger after a space a dot or something similar. Furthermore there is nothing language specific here. E.g. If you have a assembly named like this 
```
[InternalsVisibleTo("Company.ExampleAssembly, PublicKeyToken=a5$$
```
You have to do something like this:
Look up the next characters left of the current position until
* **Trigger**: you find a double quote 
* **Don't trigger**: you find an character that is not a valid assembly identifier (anything except: letters, digits, comma, dot, space, equal sign and whatever special character is allowed in an assemblyName). This has nothing to do with the `SyntaxFacts.Is...Character` family of methods.
* **Don't trigger**: You reach the start of line

Thats more like what `CommonCompletionUtilities.GetWordSpan` does, but `GetWordSpan` is more complicated because it also looks to the right (to extend the span to the end of the word). That's what I was refering to when I said 
>I looked into all the other IsInsertionTrigger implementations but I couldn't find anything that fits here.

It looks like we need to come up with something specific for this case and we need to loop to the left until the start of the line.